streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -17,116 +17,109 @@
+     * Create a new instance of {@link KStream} by transforming each element in this stream into a different element in the new stream.
      *
-     * @param mapper the instance of KeyValueMapper
-     * @param <K1>   the key type of the new stream
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link KeyValueMapper}
+     * @param <K1>          the key type of the new stream
+     * @param <V1>          the value type of the new stream
      */
     <K1, V1> KStream<K1, V1> map(KeyValueMapper<K, V, KeyValue<K1, V1>> mapper);
 
     /**
-     * Creates a new instance of KStream by transforming each value in this stream into a different value in the new stream.
+     * Create a new instance of {@link KStream} by transforming each value in this stream into a different value in the new stream.
      *
-     * @param mapper the instance of ValueMapper
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link ValueMapper}
+     * @param <V1>          the value type of the new stream
      */
     <V1> KStream<K, V1> mapValues(ValueMapper<V, V1> mapper);
 
     /**
-     * Creates a new instance of KStream by transforming each element in this stream into zero or more elements in the new stream.
+     * Create a new instance of {@link KStream} by transforming each element in this stream into zero or more elements in the new stream.
      *
-     * @param mapper the instance of KeyValueMapper
-     * @param <K1>   the key type of the new stream
-     * @param <V1>   the value type of the new stream
-     * @return the instance of KStream
+     * @param mapper        the instance of {@link KeyValueMapper}
+     * @param <K1>          the key type of the new stream
+     * @param <V1>          the value type of the new stream
      */
     <K1, V1> KStream<K1, V1> flatMap(KeyValueMapper<K, V, Iterable<KeyValue<K1, V1>>> mapper);
 
     /**
-     * Creates a new stream by transforming each value in this stream into zero or more values in the new stream.
+     * Create a new instance of {@link KStream} by transforming each value in this stream into zero or more values in the new stream.
      *
-     * @param processor the instance of Processor
-     * @param <V1>      the value type of the new stream
-     * @return the instance of KStream
+     * @param processor     the instance of {@link ValueMapper}
+     * @param <V1>          the value type of the new stream
      */
     <V1> KStream<K, V1> flatMapValues(ValueMapper<V, Iterable<V1>> processor);
 
     /**
-     * Creates an array of streams from this stream. Each stream in the array corresponds to a predicate in
+     * Create an array of {@link KStream} from this stream. Each stream in the array corresponds to a predicate in
```

Comment:
I'd rewrite to sth like:

> Creates an array of {@link KStream} from this stream by branching the elements in the original stream based on the supplied predicates.  Each element is evaluated against the supplied predicates, and predicates are evaluated in order.  Each stream in the result array corresponds position-wise (index) to the predicate in the supplied predicates.  The branching happens on first-match:  An element in the original stream is assigned to the corresponding result stream for the first predicate that evaluates to true, and assigned to this stream only.  An element will be dropped if none of the predicates evaluate to true.
>
> Example: If two predicates are supplied, then the result array will have two streams.  The first stream (at position 0) will contain all elements that match the first predicate.  The second stream (at position 1) will contain all elements that did not match the first predicate but that match the second predicate.  Any elements in the original stream that match neither of the two predicates will be dropped, i.e. they will not be included in either the first or the second result streams.

Question about corner cases: Imagine, for example, you provide 3 predicates.  What does this method return if none of the elements in the original stream match any of the predicates (= all of the elements will be dropped)?  Would the return value be `KStream[]` array of size `3`, and each of its 3 elements will be an empty `KStream`?